core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -629,6 +629,8 @@ public void run()
 
             Optional<Throwable> failure = Optional.empty();
             try {
+                // schedule() is the main logic, but expensive, so we do not want to call it after every event.
+                // Process events for some time (measured by schedulingDelayer) before invoking schedule() next time.
```

Comment:
Actually we are processing all the events from queue - and then calling schedule.
`scheduleDelayer` is only to pause calling schedule in case we observed failures which we believe may impact newly scheduled tasks too.

Reply:
Yes, `processEvents()` processes all existing events. Not even limited by EVENT_BUFFER_CAPACITY (but it does that in EVENT_BUFFER_CAPACITY portions, repeatedly)
but then, provided that `getRemainingDelayInMillis() > 0`, we call `processEvents()' waiting for new events (for 1m)